### PR TITLE
MISC: Fix Tool Settings enum validation regression and improve error messages

### PR DESCRIPTION
**Pull request by chandrasekharan-zipstack**
## What

- Fixed enum validation regression in Tool Settings where valid adapter selections were showing "'Challenger LLM' must be one of the allowed values" errors
- Enhanced form validation to properly handle adapter name-to-ID transformation

## Why

PR #1451 introduced a change where backend schemas use adapter IDs (UUIDs) for enum validation while frontend displays adapter names. This created a validation mismatch causing false validation errors even when users selected valid options.

## How

- **Adapter Name-to-ID Transformation**: Added `transformAdapterNamesToIds()` function in ToolSettings that converts adapter names to IDs during form initialization using schema enum/enumNames mapping
- **Edge Case Handling**: Added warnings for deleted/renamed adapters while maintaining graceful fallback behavior

## Can this PR break any existing features. If yes, please list possible items. If no, please explain why. (PS: Admins do not merge the PR without this section filled)

- **No breaking changes**: This is a pure bug fix that maintains backward compatibility
- **Form data transformation**: Ensures form validation works correctly with existing and new adapter data formats
- **Error message improvements**: Only enhances user experience without changing validation logic
- **Graceful degradation**: Handles edge cases (missing/renamed adapters) without breaking form functionality

## Relevant Docs

- Enhanced user experience for Tool Settings validation errors

## Related Issues or PRs

- **Related PR**: #1451 - Initial adapter ID implementation that introduced the validation regression

### Test Scenarios:
1. **Valid Selection**: Verify Tool Settings saves successfully with valid adapter selections (no false validation errors)
3. **Edge Cases**: Test with deleted/renamed adapters - should show warning but not crash
4. **Form Validation**: Ensure RJSF validateForm() works correctly with transformed data

### Manual Testing:
- Open Tool Settings for any tool with adapter configurations
- Try saving with valid selections - should work without validation errors

## Checklist

I have read and understood the [Contribution Guidelines](https://docs.unstract.com/unstract/contributing/unstract/).

### CODE REVIEW

File: frontend/src/components/agency/tool-settings/ToolSettings.jsx [hunk 1]
```diff
@@ -22,11 +22,49 @@ function ToolSettings({ spec, isSpecLoading }) {
...
+        const nameIndex = fieldSchema.enumNames.indexOf(currentValue);
+        if (nameIndex !== -1 && fieldSchema.enum[nameIndex]) {
+          // Replace name with corresponding ID from enum array
+          transformedMetadata[fieldName] = fieldSchema.enum[nameIndex];
+        } else {
+          // Handle case where adapter name is not found (possibly deleted/renamed)
+          console.warn(
+            `[ToolSettings] WARNING - Adapter '${currentValue}' for field '${fieldName}' not found in available options:`,
+            fieldSchema.enumNames
+          );
```
Comment (vishnuszipstack) [inline frontend/src/components/agency/tool-settings/ToolSettings.jsx]: @chandrasekharan-zipstack is this console.warn and else part required?

Comment (jaags-dev) [inline frontend/src/components/agency/tool-settings/ToolSettings.jsx]: Seems like we just console log the warning and not action is done here? @chandrasekharan-zipstack Why we're suppressing by logging the error on the else part? Please avoid console log.

**Review by jagadeeswaran-zipstack — CHANGES_REQUESTED** [review]

File: frontend/src/components/agency/tool-settings/ToolSettings.jsx [hunk 2]
```diff
@@ -22,11 +22,49 @@ function ToolSettings({ spec, isSpecLoading }) {
...
+          // Keep the original value, backend will handle the validation error
```
Comment (chandrasekharan-zipstack) [inline frontend/src/components/agency/tool-settings/ToolSettings.jsx]:
```suggestion
          // BE handles case where adapter name is not found (possibly deleted/renamed)
```

File: frontend/src/components/agency/tool-settings/ToolSettings.jsx [hunk 3]
```diff
@@ -22,11 +22,44 @@ function ToolSettings({ spec, isSpecLoading }) {
...
+        const currentValue = transformedMetadata[fieldName];
+
+        // Find the index of the current name in enumNames
+        const nameIndex = fieldSchema.enumNames.indexOf(currentValue);
+        if (nameIndex !== -1 && fieldSchema.enum[nameIndex]) {
+          // Replace name with corresponding ID from enum array
+          transformedMetadata[fieldName] = fieldSchema.enum[nameIndex];
+        } else {
+          // BE handles case where adapter name is not found (possibly deleted/renamed)
+        }
```
Comment (vishnuszipstack) [inline frontend/src/components/agency/tool-settings/ToolSettings.jsx]: @chandrasekharan-zipstack remove this empty else block.

**Review by vishnuszipstack — CHANGES_REQUESTED** [review]
